### DISCUSSION TITLE
Working with JobQueue

### DISCUSSION THREAD

**Discussion by Yusseiin** [python-telegram-bot/python-telegram-bot#3434] [Q&A]
Hi all, (sorry for bad eng)
i am trying to working with JobQueue in v20.0b0.
I would like to add the possibility to any user to create any kind of "reminder".
I am asking to them the time when they want to receive the message, on what day/days and what text they wanna get.
I will store everything in a TinyDB for getting back it when the job will be called.
At the end of the data collection i will create a job like this:
`context.job_queue.run_daily(remainder, chat_id=user.id, days=daysselected, time = time(hour = int(reminderhour), minute = int(reminderminute), second = 00), name=str(user.id))`

With this system i am encountering 2 problem.
The first one is that i am not considering different timezone, how can you do that? it is possible to be specified in the `time = `?

The second one is how can a user know what jobs he has active?
if i use `context.job_queue.get_jobs_by_name` i will get back `<telegram.ext._jobqueue.Job object at 0x0000027D9092FB80>` how can you get back more information ? like days and time of the jobqueue?
It is possible to get the same information that i am getting in the console?
`apscheduler.executors.default - INFO - Running job "reminder (trigger: date[2022-12-20 14:07:38 UTC], next run at: 2022-12-20 14:07:38 UTC)" (scheduled at 2022-12-20 14:07:38.186185+00:00)`

Thanks to everyone

**Comment by Bibo-Joshi** [comment]
timezones: Either make the time timezone aware or use [`Defaults`](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Adding-defaults-to-your-bot) to set a default timezone for your bot.

Getting info about jobs: Please see the docs of the [`Job`](https://docs.python-telegram-bot.org/en/v20.0b0/telegram.ext.job.html#telegram-ext-job) class for the details on that class.

**Reply by Yusseiin** [reply]
timezones: if i set the timezone on my bot it won't change for the user. If i have an user from US and a user from EU my bot won't recognize the timezone difference. It is possible to do so?

On the docs of the Job class i can not find any info on how to get the job that are running on my bot.

**Reply by Bibo-Joshi** [reply]
the only reliable way to know which timezone a user is in is to ask them.

These properties give you the information that you can see in the log entry:

* https://docs.python-telegram-bot.org/en/v20.0b0/telegram.ext.job.html#telegram.ext.Job.next_t
* https://docs.python-telegram-bot.org/en/v20.0b0/telegram.ext.job.html#telegram.ext.Job.job

**Reply by Yusseiin** [reply]
I understand that i have to ask the user, but after how can i setup correctly `context.job_queue.run_daily`?
I need to change hour/minute of the job based on timezone or there is a better way to do so?

**Reply by Bibo-Joshi** [reply]
you can make your time objects timezone aware. That's a rather general time/datetime thing though and not specific to python-telegram-bot.
